bazaar/interface.rst

```diff
@@ -1,12 +1,207 @@
 
-Bazaar Interfaces
-*****************
+Edustore Distributed API
+=========================
 
-LMS
-===
+draft v0.1
 
+.. contents::
 
-CMS
-===
 
+Signing and Authentication 
+==========================
 
+There are two supported methods:
+
+1. Header HMAC256 AUTH token in headers and JSON body (Similar: http://docs.aws.amazon.com/AmazonS3/latest/dev/RESTAuthentication.html)
+2. GET/POST request with params=<json>&authentication=<client_id:hash>
+
+1 Adding material
+==============================
+
+
+1.1 Searching and adding material to the LMS
+--------------------------------------------
+
+Learning material can be added with three use methods:
+
+1. User open Edustore repository directly.
+2. | User select materials from the Edustore repository to the LMS by using: 
+   | - Drag and Drop
+   | - Window to window
+   | NOTE: These material selection methods cannot be used from mobile devices (Windows Phone, iOS, Android). 
+3. User selects material from integrated embedded Edustore JS module, which can be included to any LMS. This method uses drag and drop.
+
+
+API workflow for selecting material from Edustore directly:
+
+1. LMS makes a JSON POST request to the Edustore with params from table 1, using header-based auth.
+2. Edustore replies with unique URL and LMS forwards user to the unique url.  
```

Comment:
Specification of the reply is missing. I assume it comes in the body of the response, but how it is encoded? I assume UTF-8 again, but is it inside JSON document like the request was?